lib/ansible/cli/doc.py

```diff
@@ -129,18 +129,23 @@ def run(self):
 
         # process all plugins of type
         if self.options.all_plugins:
-            self.args = self.get_all_plugins_of_type(plugin_type, loader)
+            self.args = self.get_all_plugins_of_type(plugin_type)
 
-        # dump plugin metadata as JSON
+        # dump plugin desc/metadata as JSON
         if self.options.json_dump:
             plugin_data = {}
-            for plugin_type in C.DOCUMENTABLE_PLUGINS:
+            if not plugin_type:
+                plugins = C.DOCUMENTABLE_PLUGINS
+            else:
+                plugins = [plugin_type]
```

Comment:
This won't work as the default of plugin_type is module in the OptParse parser so we'll never hit the not plugin_type path.

Reply:
do you actually use all plugins always in json option? i was thinking it could be a good way for other tools to query ansible 'what become plugins do you have'?